### ISSUE TITLE
Running clipper on Encode data: ERROR:root:peak finding failed

### ISSUE THREAD

**Issue by tefina** [YeoLab/clipper#64]
Hi,

I installed clipper from the encode branch and tried to process an alignment file from the Encode project (Experiment ENCSR862QCH, hg19 alignment, replicate 2, used file: ENCFF367JCV.bam).
I sorted the file and extracted read 2 according to the instructions in the attached eCLIP analysis protocol. I ran clipper as specified in the protocol.

`clipper -b ENCFF367JCV_sort_r2.bam -s hg19 -o peaks.bed --bonferroni -- superlocal --threshold-method binomial --save-pickle`

Clipper runs but shows many error messages:
ERROR:root:peak finding failed:, ENSG00000210077.1, boolean index did not match indexed array along dimension 0; dimension is 54 but corresponding boolean dimension is 23
ERROR:root:peak finding failed:, ENSG00000210100.1, boolean index did not match indexed array along dimension 0; dimension is 44 but corresponding boolean dimension is 29
ERROR:root:peak finding failed:, ENSG00000198763.3, boolean index did not match indexed array along dimension 0; dimension is 265 but corresponding boolean dimension is 35
ERROR:root:peak finding failed:, ENSG00000198888.2, boolean index did not match indexed array along dimension 0; dimension is 499 but corresponding boolean dimension is 46

In the end I get an output file with very few peaks (about 30). 
I also tried clipper on our own data and run in the same set of error messages.

I would appreciate any help, not sure what's going on..

Best,
Stefanie

**Comment by yuxinghai** [comment]
I met the same problem, anyone could help ?

**Comment by byee4** [comment]
In the master branch, the options --bonforroni, --superlocal, and --threshold-method are removed as they run by default now. Try with minimal parameters 
```
clipper \
-s hg19 \
-o ENCFF367JCV.peaks.bed \
-b ENCFF367JCV.sorted.r2.bam
```

**Comment by madaojjc** [comment]
I met the same problem.. Have you solved this issue?

**Comment by cainbn97** [comment]
I believe this is a numpy deprecation error. 

[https://stackoverflow.com/questions/45207650/indexerror-boolean-index-did-not-match-indexed-array-along-dimension-0](url)

I removed the `numpy==1.16.6` that was installed with CLIPper and installed `numpy==1.10.2`. These errors turned into the warning below, but CLIPper ran fine.

`WARNING: DeprecationWarning: numpy boolean subtract, the `-` operator, is deprecated, use the bitwise_xor, the `^` operator, or the logical_xor function instead.
  return a[slice1]-a[slice2]`

I hope this helps!

**Comment by madaojjc** [comment]
Thanks a lot ! I will try it !




















At 2020-06-30 03:42:15, "cainbn97" <notifications@github.com> wrote:

I believe this is a numpy deprecation error.

https://stackoverflow.com/questions/45207650/indexerror-boolean-index-did-not-match-indexed-array-along-dimension-0

I removed the numpy==1.16.6 that was installed with CLIPper and installed numpy==1.10.2. These errors turned into the warning below, but CLIPper ran fine.

WARNING: DeprecationWarning: numpy boolean subtract, the -operator, is deprecated, use the bitwise_xor, the^ operator, or the logical_xor function instead. return a[slice1]-a[slice2]

I hope this helps!

**Comment by algaebrown** [comment]
The deprecation error should be solved with new clipper release 2.0.0
